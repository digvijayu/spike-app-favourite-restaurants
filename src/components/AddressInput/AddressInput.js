import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import RestaurantSelectionList from './../RestaurantSelectionList';

const InputContainer = styled.div``;

const Input = styled.input`
  -webkit-transition: all 0.3s ease-in-out;
  -moz-transition: all 0.3s ease-in-out;
  -ms-transition: all 0.3s ease-in-out;
  -o-transition: all 0.3s ease-in-out;
  outline: none;
  box-sizing: border-box;
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  width: 100%;
  background: #fff;
  border: 2px solid #ccc;
  padding: 0.5rem;
  color: #555;
  font-size: 1rem;

  &:hover {
    border: 2px solid #aaa;
  }

  &:focussed {
    border: 2px solid #aaa;
  }
`;

class AddressInput extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: props.name
    };
  }

  handleOnNameChange(e) {
    this.props.onChange(e.target.value);
  }

  render() {
    const { name } = this.state;
    return (
      <InputContainer>
        <Input
          className="AddressForm__name-input"
          type="text"
          value={name}
          onChange={this.handleOnNameChange.bind(this)}
        />
        <RestaurantSelectionList restaurants={[]} />
      </InputContainer>
    );
  }
}

AddressInput.propTypes = {
  name: PropTypes.string,
  onChange: PropTypes.func.isRequired
};

export default AddressInput;
