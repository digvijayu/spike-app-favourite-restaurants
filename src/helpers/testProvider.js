import React from 'react';
import { Provider } from 'react-redux';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from './../store';

configure({ adapter: new Adapter() });

function testProvider(component) {
  const { store } = configureStore();
  const jsx = (
    <Provider store={store}>{React.cloneElement(component)}</Provider>
  );
  return {
    jsx,
    store
  };
}

export default testProvider;
