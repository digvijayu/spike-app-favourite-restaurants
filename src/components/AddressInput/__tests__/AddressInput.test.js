import React from 'react';
import reactRenderer from 'react-test-renderer';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AddressInput from './../AddressInput';
import testProvider from './../../../helpers/testProvider';

configure({ adapter: new Adapter() });

it('should render the address input', () => {
  const cb = jest.fn();
  const tree = reactRenderer
    .create(<AddressInput name={'Mock Restaurant Name'} onChange={cb} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('should invoke on change when changed', () => {
  const cb = jest.fn();
  const input = mount(
    <AddressInput name={'Mock Restaurant Name'} onChange={cb} />
  );
  input.find('input').simulate('change', { target: { value: 'newVal' } });
  expect(cb.mock.calls.length).toBe(1);
  expect(cb.mock.calls[0][0]).toBe('newVal');
});
