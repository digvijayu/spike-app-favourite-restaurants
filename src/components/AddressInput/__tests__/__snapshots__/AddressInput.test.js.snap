// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the address input 1`] = `
<div
  className="sc-bxivhb eBNJKg"
>
  <input
    className="AddressForm__name-input sc-ifAKCX dEXFVu"
    onChange={[Function]}
    type="text"
    value="Mock Restaurant Name"
  />
  <div
    className="sc-bdVaJa hGdLZl"
  >
    <div
      className="sc-htpNat iDoDco"
    >
      No items available.
    </div>
  </div>
</div>
`;
